src/analysis/processing/qgsalgorithmconcavehull.cpp

```diff
@@ -0,0 +1,250 @@
+/***************************************************************************
+                         qgsalgorithmconcavehull.cpp
+                         ---------------------
+    begin                : July 2023
+    copyright            : (C) 2023 by Alexander Bruy
+    email                : alexander dot bruy at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmconcavehull.h"
+#include "qgsspatialindex.h"
+#include "qgsmultipoint.h"
+#include "qgsprocessingregistry.h"
+#include "qgsvectorlayer.h"
+
+///@cond PRIVATE
+
+QString QgsConcaveHullAlgorithm::name() const
+{
+  return QStringLiteral( "concavehull" );
+}
+
+QString QgsConcaveHullAlgorithm::displayName() const
+{
+  return QObject::tr( "Concave hull" );
+}
+
+QStringList QgsConcaveHullAlgorithm::tags() const
+{
+  return QObject::tr( "concave,hull,bounds,bounding" ).split( ',' );
+}
+
+QString QgsConcaveHullAlgorithm::group() const
+{
+  return QObject::tr( "Vector geometry" );
+}
+
+QString QgsConcaveHullAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorgeometry" );
+}
+
+QString QgsConcaveHullAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "computes the concave hull of the features from an input layer." );
+}
+
+QgsConcaveHullAlgorithm *QgsConcaveHullAlgorithm::createInstance() const
+{
+  return new QgsConcaveHullAlgorithm();
+}
+
+void QgsConcaveHullAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ), QObject::tr( "Input layer" ), QList< int >() << QgsProcessing::TypeVectorPoint ) );
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "ALPHA" ), QObject::tr( "Threshold" ), QgsProcessingParameterNumber::Double, 0.3, false, 0, 1 ) );
```

Comment:
The extra "Threshold (0-1, where 1 is equivalent with Convex Hull)" help text could be added as the help for this parameter